tests/test_domain_objects/cashflow_test.py

```diff
@@ -0,0 +1,43 @@
+import sys
+from test_domain_objects import shared_tests as shared
+from test_domain_objects import helper_methods as helper
+
+
+sys.path.insert(0, 'tests/')
+
+
+def test_cashflows():
+    """ Ensure all generated cashflow attributes adhere to their
+    specification. Dependent on Swap Positions """
+
+    records = helper.set_up_cashflow_tests()
+    for record in records:
+        shared.swap_contract_id_exists(record)
+        shared.ric_exists(record)
+        cashflow_type_exists(record)
+        # pay_date_valid(record)
+        effective_date_exists(record)
+        shared.currency_valid(record)
+        shared.amount_valid(record)
+        shared.long_short_valid(record)
+
+
+def effective_date_exists(record):
+    formatted_effective_date = record['effective_date'].strftime("%Y-%m-%d")
```

Comment:
Date format should be pulled from the CashFlow (factory) class so that the test doesn't fail if the format is changed

Reply:
ISSUE

Reply:
https://github.com/galatea-associates/fuse-test-data-gen/issues/66